Neos.ContentRepository.Core/Classes/Projection/ContentGraph/AbsoluteNodePath.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Projection\ContentGraph;
+
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Exception\AbsoluteNodePathIsInvalid;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeName;
+
+/**
+ * An absolute node path, composed of the root node's type and a relative path from there.
+ *
+ * It describes the hierarchy path of a node to and including its root node in a subgraph.
+ * @api
+ */
+final class AbsoluteNodePath implements \JsonSerializable
+{
+    private function __construct(
+        public readonly NodeTypeName $rootNodeTypeName,
+        public readonly NodePath $path
+    ) {
+    }
+
+    public static function fromComponents(
+        NodeTypeName $rootNodeTypeName,
+        NodePath $path
+    ): self {
+        return new self(
+            $rootNodeTypeName,
+            $path
+        );
+    }
+
+    public static function fromString(string $value): self
+    {
+        if (\str_starts_with($value, '/<')) {
+            $pivot = \mb_strpos($value, '>');
+            if ($pivot > 0) {
+                $nodeTypeName = NodeTypeName::fromString(\mb_substr($value, 2, $pivot - 2));
+                $path = \mb_substr($value, $pivot + 2) ?: '/';
+            } else {
+                throw AbsoluteNodePathIsInvalid::becauseItDoesNotMatchTheRequiredPattern($value);
+            }
+        } else {
+            throw AbsoluteNodePathIsInvalid::becauseItDoesNotMatchTheRequiredPattern($value);
+        }
```

Comment:
Just a cosmetic nitpick, but an early return would save some cognitive load:
```suggestion
        if (!\str_starts_with($value, '/<')) {
            throw AbsoluteNodePathIsInvalid::becauseItDoesNotMatchTheRequiredPattern($value);
        }
        $pivot = \mb_strpos($value, '>');
        if ($pivot === false) {
            throw AbsoluteNodePathIsInvalid::becauseItDoesNotMatchTheRequiredPattern($value);
        }
        $nodeTypeName = NodeTypeName::fromString(\mb_substr($value, 2, $pivot - 2));
        $path = \mb_substr($value, $pivot + 2) ?: '/';
```

Reply:
i plan to refactor this to regex either way ^^

in the same run as fixing the last remaining unstrict dtos: https://github.com/neos/neos-development-collection/issues/4133